designer/client/src/components/modals/ValidationLabels.tsx

```diff
@@ -22,22 +25,33 @@ const LabelsContainer = styled.div({
 
 export default function ValidationLabels(props: Props) {
     const { validators, values, validationLabelInfo } = props;
+    const { t } = useTranslation();
 
     const validationErrors: ValidationError[] = withoutDuplications(validators)
         .filter((v) => !v.isValid(...values))
         .map((validator) => ({
             message: validator.message && validator.message(),
             description: validator.description && validator.description(),
+            templateValues: validator.templateValues && validator.templateValues(),
+            errorCode: validator.errorCode && validator.errorCode(),
         }));
 
     const isValid: boolean = isEmpty(validationErrors);
 
     const renderErrorLabels = () =>
         validationErrors.map((validationError) => {
+            const { errorCode, templateValues, message } = validationError;
+
+            const canRenderI18nTemplate = errorCode != null && templateValues != null;
+
             // we don't pass description as tooltip message until we make changes on the backend
             return (
-                <LimitedValidationLabel key={validationError.message} title={validationError.message} type="ERROR">
-                    {validationError.message}
+                <LimitedValidationLabel key={message} type="ERROR" title={message}>
+                    {canRenderI18nTemplate ? (
+                        <I18nTemplateLabel errorCode={errorCode} templateValues={validationError.templateValues} />
```

Comment:
How many template variants we will have? I checked `designer/client/src/types/definition.ts` but I'm not able to determine it, based on these types. Maybe it will be easier, if we try to write down, what variants of validation errors we have in our system

Reply:
Right now it's a mess :/ - I'm still trying to figure out how many are needed and right now I think we should have 2: a simple string value and typingResult value. The string could be interpolated without having a dedicated component, but the typingResult requires a custom span component. 

I am working on having a union type with a template and non-templated error - I'm hoping this will be a valid solution.